src/app/compiler.js

```diff
@@ -102,14 +102,14 @@ function Compiler (editor, queryParams, handleGithubCall, updateFiles) {
     var noFatalErrors = true; // ie warnings are ok
 
     if (data['error'] !== undefined) {
-      self.event.trigger('compilationError', [data['error']]);
+      self.event.trigger('compilationFinished', [false, data['error'], editor.getValue()]);
       if (utils.errortype(data['error']) !== 'warning') {
         noFatalErrors = false;
       }
     }
     if (data['errors'] !== undefined) {
       data['errors'].forEach(function (err) {
-        self.event.trigger('compilationError', [err]);
+        self.event.trigger('compilationFinished', [false, err, editor.getValue()]);
```

Comment:
I thought `compilationError` was for just reporting a single error? `compilationFinished` should only be triggered once per compilation run.